File: programa.mjs

import * as readline from 'node:readline/promises';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let personas = [];

while (true) {
    let nombre = await rl.question('Ingrese el nombre de la persona (o escriba "finalizar" para terminar): ');
    if (nombre.toLowerCase() === 'finalizar')
        break; 

let edad = parseInt(await rl.question ("Ingrese la edad de la persona:"));
let nota = parseFloat(await rl.question ("Ingrese la nota de la persona:"));

personas.push([nombre, edad, nota]);

}

console.log("Lista de personas:" , personas);

rl.close

let listaordena = [...personas].sort((a, b) => b[2] - a[2]);

console.log("Lista ordenada:" , listaordena);

File: programa.py

personas = [] 

while ( True ): 
    nombre = input("Ingrese el nombre de la persona(o escriba 'finalizar' para terminar): ")
    if nombre == "finalizar":
        break 

    edad = int(input("ingrese la edad de la persona:"))    
    nota = float(input('ingrese la nota de la persona:'))

    lista = [ nombre, edad, nota]
    personas.append(lista)

    print (personas)

    lista_ordenada = sorted(personas, key=lambda posicion: posicion[2], reverse=True)

    print (lista_ordenada)

    suma_notas = 0 
    for posicion in lista_ordenada:
        suma_notas += posicion[2]

    promedio = suma_notas / len(lista_ordenada)
    
    print ( "El promedio de las notas es:", promedio)

